### PR TITLE
Performance Improvements

### PR DESCRIPTION
**Pull request by gblakeman**
A pass at some performance enhancements:

* Chains some node filtering into a single `findAll` call.
* Switches a few `filter` and `forEach` loops into `for` with `break` when iterating the entire array is not needed (questionable improvements given how small our array sizes usually are, but it makes me feel better).
* Removes some unnecessary functions (like `retrieveNode`) where Figma has better API-level alternatives.
* Removes the need to traverse `main` components and read their data. This is our biggest win. I forget when Figma changed the data inheritance on instance nodes, but this code was both incredibly slow and doing double-duty.

### CODE REVIEW

File: src/App.ts [hunk 1]
```diff
@@ -68,39 +68,10 @@ const getFilteredNodes = (
   const filterTypes: Array<
     ('COMPONENT' | 'ELLIPSE' | 'FRAME' | 'INSTANCE' | 'POLYGON' | 'RECTANGLE' | 'STAR' | 'TEXT')
   > = ['COMPONENT', 'ELLIPSE', 'FRAME', 'INSTANCE', 'POLYGON', 'RECTANGLE', 'STAR', 'TEXT'];
-  const typeFilteredNodes: Array<RealishFilteredNodes> = new Crawler(
+  const filteredNodes: Array<RealishFilteredNodes> = new Crawler(
     { for: consolidatedSelection },
   ).filterByTypes(filterTypes);
 
-  const filteredNodes: Array<RealishFilteredNodes> = [];
```
Comment (gblakeman) [inline src/App.ts:75]: This stuff all got moved into `filterByTypes` in `Crawler` (or eliminated altogether in some cases).

File: src/App.ts [hunk 2]
```diff
@@ -514,30 +492,9 @@ export default class App {
...
-    /**
-     * Filters the available `selection` and finds a text node (`TextNode`)
-     * matching the `id` provided in the `payload`.
-     *
-     * @kind function
-     * @name retrieveNode
-     *
-     * @returns {Object} The node (`RealishFilteredNodes`) retrieved.
-     */
-    const retrieveNode = (): RealishFilteredNodes => {
```
Comment (gblakeman) [inline src/App.ts:529]: I don’t remember if `figma.getByNodeId` was not in the API originally, or if I was still so green with the API that I thought I had to do this manually… 🤦🏻‍♂️

**Review by joeyyang — APPROVED** [review]
LGTM!

**Review by lynnerang — APPROVED** [review]
Nice!

**Review by lynnerang — APPROVED** [review]
Looks like a real nice refactor :D

**Pull request merged**